monitoring/kernel_linux_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring_test
```

Comment:
Hmm - why is this a different package than `monitoring`?

Reply:
The linter was complaining ```monitoring/kernel_linux_test.go:128:9: package should be `monitoring_test` instead of `monitoring` (testpackage)
package monitoring```. New warning might be because I upgraded the golangci-lint version a little while ago?

I'm not sure why it recommends this test file to be under `monitoring_test` package, but not other test files.

Reply:
I've never seen this before hence I asked. It'd make sense for black box testing, yes, but not when  the exports are solely composed from other (possibly non-exported ones) which are then unit-tested. The composed functions are then by definition tested if the lower functions they're composed of are tested and the exports do not do any other logic.

Reply:
That makes sense. The parsing function was initially un-exported anyways, and it doesn't make sense to export it just for the sake of fitting it in a black box test. 

I'll update the kernel_checker tests to test the results of `Check`, which should be more closer to an integration test.